### PR TITLE
[rustdoc] Fix `--show-coverage` when JSON output format is used

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
I realized while looking on the docs.rs page of the `sysinfo` crate that the coverage numbers displayed were wrong:

image

I realized that it was because `--show-coverage --output-format=json` was relying on the same logic as the JSON output for the doc generation whereas it should not. I fixed it by changing the API for querying `is_json` a bit.

The underlying issue is that JSON output format is stripping reexports of items from private modules.

r? @notriddle

### CODE REVIEW

File: src/librustdoc/core.rs [hunk 1]
```diff
@@ -121,6 +121,13 @@ impl<'tcx> DocContext<'tcx> {
             _ => None,
         }
     }
+
+    /// Returns `true` if the JSON output format is enabled for generating the crate content.
+    ///
+    /// If another option like `--show-coverage` is enabled, it will return false.
+    pub(crate) fn is_json(&self) -> bool {
```
Comment (notriddle) [inline src/librustdoc/core.rs:128]:
```suggestion
    pub(crate) fn is_json_output(&self) -> bool {
```

Comment (GuillaumeGomez) [reply src/librustdoc/core.rs]: Done!

**Comment by GuillaumeGomez** [comment]
I somehow successfully failed at committing all changes. ^^'

**Pull request merged**